src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -420,6 +421,44 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                             methodBuilder.Add(method);
                         }
                     }
+                    else
+                    // argumentsOnly == true
+                    {
+                        foreach(var argumentName in _argumentNames)
+                        {
+                            bool found = false;
+                            int localIndex = 0;
+                            foreach (var local in _localsForBinding)
+                            {
+                                if (local.Name == argumentName)
+                                {
+                                    AppendLocalAndMethod(localBuilder, methodBuilder, local, container, localIndex, GetLocalResultFlags(local));
+                                    found = true;
+                                    break;
+                                }
+                                localIndex++;
+                            }
+
+                            if (!found)
+                            {
+                                // Method parameters (except those that have been hoisted).
+                                int parameterIndex = m.IsStatic ? 0 : 1;
+                                foreach (var parameter in m.Parameters)
+                                {
+                                    var parameterName = parameter.Name;
+                                    if (parameterName == argumentName &&
+                                        GeneratedNames.GetKind(parameterName) == GeneratedNameKind.None &&
+                                        !IsDisplayClassParameter(parameter))
+                                    {
+                                        AppendParameterAndMethod(localBuilder, methodBuilder, parameter, container, parameterIndex);
+                                        break;
+                                    }
+
+                                    parameterIndex++;
+                                }
+                            }
+                        }
+                    }
```

Comment:
There are two completely independent paths: 
For !argumentsOnly, we display pseudo-variables, then, hoisted parameters, then other parameters and then locals.  For arguments only, we display just arguments and in a rigid order.

I do not see that we can merge those paths.

---
In reply to: [134929193](https://github.com/dotnet/roslyn/pull/21707#discussion_r134929193) [](ancestors = 134929193)